flutter/lib/src/sentry_asset_bundle.dart

```diff
@@ -244,4 +285,41 @@ class SentryAssetBundle implements AssetBundle {
 
     return data;
   }
+
+  static FutureOr<T> _wrapBinaryParsing<T>(
+    _ByteParser<T> parser,
+    ByteData value,
+    String key,
+    ISentrySpan? outerSpan,
+  ) async {
+    final span = outerSpan?.startChild(
+      'serialize.file.read',
+      description: 'parsing "$key" to "$T"',
+    );
+    T data;
+    try {
+      data = await parser(value);
+      span?.status = const SpanStatus.ok();
+    } catch (e) {
+      span?.throwable = e;
+      span?.status = const SpanStatus.internalError();
+      rethrow;
+    } finally {
+      await span?.finish();
+    }
+
+    return data;
+  }
+
+  @override
+  // ignore: override_on_non_overriding_member
+  Future<T> loadStructuredBinaryData<T>(
+    String key,
+    FutureOr<T> Function(ByteData data) parser,
+  ) async {
+    if (_enableStructuredDataTracing) {
+      return _loadStructuredBinaryDataWithTracing(key, parser);
+    }
+    return _bundle.loadStructuredBinaryData(key, parser);
+  }
```

Comment:
```suggestion
  @override
  // ignore: override_on_non_overriding_member
  Future<T> loadStructuredBinaryData<T>(
    String key,
    FutureOr<T> Function(ByteData data) parser,
  ) async {
    if (_enableStructuredDataTracing) {
      return _loadStructuredBinaryDataWithTracing(key, parser);
    }
    
    return _loadStructuredBinaryDataWrapper(key, parser); 
  }
  
  // helper method to have a "typesafe" method
  Future<T> _loadStructuredBinaryDataWrapper(
    String key,
    FutureOr<T> Function(ByteData data) parser,
  ) async {
      // The loadStructuredBinaryData method exists as of Flutter greater than 3.8
      // Previous versions don't have it, but later versions do.
      // We can't use `extends` in order to provide this method because this is
      // a wrapper and thus the method call must be forwarded.
      // On Flutter versions <=3.8 we can't forward this call.
      // On later version the call gets correctly forwarded.
      // The error doesn't need to handled since it can't be called on earlier versions,
      // and it's correctly forwareded on later versions.
      return (_bundle as dynamic).loadStructuredBinaryData(key, parser) as Future<T>; 
  }
```